### ISSUE TITLE
auxiliaryCommentAfter placed after next comment

### ISSUE THREAD

**Issue by kpdecker** [babel/babel#3034]
Output:

```
  var store = /*istanbul ignore start*/(0, _redux. /*istanbul ignore end*/applyMiddleware)(
        /*istanbul ignore start*/_reducers2.default);

  /* my comment */
  /*istanbul ignore end*/
  if (module.hot) {
```

Expected:

```
  var store = /*istanbul ignore start*/(0, _redux. /*istanbul ignore end*/applyMiddleware)(
        _reducers2/*istanbul ignore start*/.default/*istanbul ignore end*/);

  /* my comment */
  if (module.hot) {
```

From:

```
  let store = applyMiddleware(reducers);

  /* my comment */
  if (module.hot) {
```

Subset of module versions:

```
├─┬ babel-loader@6.1.0
├─┬ babel-core@6.1.2
├─┬ babel-preset-es2015@6.1.2
│ ├─┬ babel-plugin-transform-es2015-modules-commonjs@6.1.3
```

Config:

```
  sourceMap: 'inline',
  auxiliaryCommentBefore: 'istanbul ignore start',
  auxiliaryCommentAfter: 'istanbul ignore end',
  presets: [
    require.resolve('babel-preset-react'),
    require.resolve('babel-preset-es2015'),
    require.resolve('babel-preset-stage-0')
  ]
```

**Comment by sebmck** [comment]
Can you be more specific? What options are you using? What output are you getting and what output do you expect?

**Comment by kpdecker** [comment]
Updated inline. Note that the exact placement of the comments for default lookup doesn't have a clear right answer to me from the discussion in #2757 but extending past the `);` seems incorrect and the problematic part for coverage flagging is it extending past the in source `/* my comment */` string.

**Comment by kpdecker** [comment]
Also manually added whitespace to the generated and expected that are not there in the generated output to try to make the difference in the comment handling clearer.

**Comment by kpdecker** [comment]
There are a number of things that are influencing this:

1. The comment above is treated as a trailing comment to the call expression. Since the aux after is not printed until the before next node is printed, this caused the source comment to print before the aux comment.
2. Aux after only prints when entering a node. Since the default lookup is the terminal parameter which is also the terminal leaf node, nothing emits this until the next node is entered, which will be outside of the `);` that is source code.

I have two potential fixes (branch is based on #3050 so can't PR just yet), 38dd571 is a quick fix for just (1) which resolves the root issue but doesn't perform 100% correctly WRT structural content like `(`, etc. 3172ac6 provides an example of how this can be fixed as well as the impact that this will have on the output. The downside here is that this needs to be done for every node that manually pushes content onto the printer buffer, or some sort of `pushLiteral` helper needs to be created to perform this operation.

Glad to continue through making this work since coverage is vital to my development flow but wanted to get a feeling for which solution was preferred.

**Comment by loganfsmyth** [comment]
I'm just going to close this. auxiliaryCommentBefore is flaky at absolute best. If you need coverage reporting via Instanbul, use istanbuljs/babel-plugin-istanbul
